rancher2/structure_node_template_azure_test.go

```diff
@@ -0,0 +1,192 @@
+		"size":                       "large",
+		"ssh_user":                   "user",
+		"static_public_ip":           false,
+		"storage_type":               "hdd",
+		"subnet":                     "1.1.1.1",
+		"subnet_prefix":              "1.1.1.1",
+		"subscription_id":            "sub-id",
+		"update_domain_count":        "3",
+		"use_private_ip":             false,
+		"vnet":                       "test-vnet",
+		"tags":                       "key1,value1",
+		"use_public_ip_standard_sku": true,
+		"availability_zone":          "1",
+		"accelerated_networking":     true,
+	}
+	testAzureNodeTemplateAnnotationsConf := map[string]string{
+		"key": "value",
+	}
+	testAzureNodeTemplateAnnotationsInterface := map[string]interface{}{
+		"key": "value",
+	}
+
+	useInternalIP := false
+	testAzureNodeTemplateConf = &NodeTemplate{
+		NodeTemplate: managementClient.NodeTemplate{
+			Driver:               "azure",
+			UseInternalIPAddress: &useInternalIP,
+			Annotations:          testAzureNodeTemplateAnnotationsConf,
+			CloudCredentialID:    "abc-test-123",
+			NodeTaints:           testAzureNodeTemplateNodeTaintsConf,
+			EngineInstallURL:     "http://fake.url",
+			Name:                 "test-node-template",
+		},
+		AzureConfig: &testNodeTemplateAzureConf,
+	}
+
+	testNodeTemplateSquashAzureConfInterface = map[string]interface{}{
+		"annotations":             testAzureNodeTemplateAnnotationsInterface,
+		"driver":                  "azure",
+		"cloud_credential_id":     "abc-test-123",
+		"use_internal_ip_address": useInternalIP,
+		"engine_install_url":      "http://fake.url",
+		"name":                    "test-node-template",
+	}
+
+	testNodeTemplateExpandAzureConfInterface = map[string]interface{}{
+		"annotations":             testAzureNodeTemplateAnnotationsInterface,
+		"node_taints":             testAzureNodeTemplateNodeTaintsInterface,
+		"driver":                  "azure",
+		"cloud_credential_id":     "abc-test-123",
+		"use_internal_ip_address": useInternalIP,
+		"engine_install_url":      "http://fake.url",
+		"name":                    "test-node-template",
+		"azure_config":            []interface{}{testAzureNodeTemplateInterface},
+	}
+}
+
+func TestFlattenAzureNodeTemplate(t *testing.T) {
+	cases := []struct {
+		Input          *NodeTemplate
+		ExpectedOutput map[string]interface{}
+	}{
+		{
+			testAzureNodeTemplateConf,
+			testNodeTemplateSquashAzureConfInterface,
+		},
+	}
+
+	for _, tc := range cases {
+		output := schema.TestResourceDataRaw(t, nodeTemplateFields(), map[string]interface{}{})
+		err := flattenNodeTemplate(output, tc.Input)
+		if err != nil {
+			t.Fatalf("[ERROR] on flattener: %#v", err)
+		}
+		expectedOutput := map[string]interface{}{}
+		for k := range tc.ExpectedOutput {
+			expectedOutput[k] = output.Get(k)
+		}
+		if !reflect.DeepEqual(expectedOutput, tc.ExpectedOutput) {
+			t.Fatalf("Unexpected output from flattener. \nExpected: %#v\nGiven:::: %#v", expectedOutput, tc.ExpectedOutput)
```

Comment:
```suggestion
			t.Fatalf("Unexpected output from flattener.\nExpected: %#v\nGiven: %#v", expectedOutput, tc.ExpectedOutput)
```
or
    
```suggestion
			t.Fatalf("Unexpected output from flattener.\nExpected: %#v\nGiven:    %#v", expectedOutput, tc.ExpectedOutput)
```
They seem to be used interchangeably.

Reply:
Oh, that must have sneaked back in during some resets while testing the build issues